src/tests/IntegrationTests/Events/RepositoryManagerTests.cs

```diff
@@ -504,7 +505,64 @@ await RepositoryManager.SwitchBranch("branch2")
             repositoryManagerListener.DidNotReceive().OnLocksUpdated(Args.EnumerableGitLock);
         }
 
-        private void WaitForNotBusy(RepositoryManagerEvents managerEvents, int seconds = 1)
+		[Test]
+		public async Task ShouldCheckoutFiles()
+		{
+			await Initialize( TestRepoMasterCleanSynchronized );
+
+			var repositoryManagerListener = Substitute.For<IRepositoryManagerListener>();
+			repositoryManagerListener.AttachListener( RepositoryManager, repositoryManagerEvents );
+
+			var expected = new GitStatus
+			{
+				Behind = 1,
+				LocalBranch = "master",
+				RemoteBranch = "origin/master",
+				Entries =
+					new List<GitStatusEntry> {
+						new GitStatusEntry("foobar.txt", TestRepoMasterCleanSynchronized.Combine("foobar.txt"),
+							"foobar.txt", GitFileStatus.Untracked)
+					}
+			};
+
+			var result = new GitStatus();
+			Environment.Repository.OnStatusUpdated += status => { result = status; };
+
+			var foobarTxt = TestRepoMasterCleanSynchronized.Combine( "foobar.txt" );
+			foobarTxt.WriteAllText( "foobar" );
```

Comment:
Okay... so far so good up until this point..

Just to reiterate what we have..
We have a test environment using a clean repository and you added a file to it `foobar.txt` and you wrote to that file.